File: app/src/main/java/com/softwareverde/tidyduck/most/Operation.java
package com.softwareverde.tidyduck.most;

public class Operation {
    private Long _id;
    private String _name;
    private String _opcode;
    private boolean _isInput;
    private String _channel = com.softwareverde.mostadapter.Operation.DEFAULT_OPERATION;

    public Long getId() {
        return _id;
    }

    public void setId(Long id) {
        _id = id;
    }

    public String getName() {
        return _name;
    }

    public void setName(String name) {
        _name = name;
    }

    public String getOpcode() {
        return _opcode;
    }

    public void setOpcode(String opcode) {
        _opcode = opcode;
    }

    public boolean isInput() {
        return _isInput;
    }

    public void setInput(boolean input) {
        _isInput = input;
    }

    public String getChannel() {
        return _channel;
    }

    public void setChannel(final String channel) {
        _channel = channel;
    }
}

File: app/src/main/webapp/js/classes/operation.js
class Operation {
    static fromJson(json) {
        const operation = new Operation();

        operation.setId(json.id);
        operation.setName(json.name);
        operation.setChannel(json.channel);

        return operation;
    }

    static toJson(operation) {
        return {
            id:         operation.getId(),
            name:       operation.getName(),
            channel:    operation.getChannel()
        };
    }

    constructor() {
        this._id        = null;
        this._name      = null;
        this._channel   = "Control";
    }

    setId(id) {
        this._id = id;
    }

    getId() {
        return this._id;
    }

    setName(name) {
        this._name = name;
    }

    getName() {
        return this._name;
    }

    setChannel(channel) {
        this._channel = channel;
    }

    getChannel() {
        return this._channel;
    }
}